karate-core/src/main/java/com/intuit/karate/http/HttpClient.java

```diff
@@ -169,7 +169,7 @@ private T buildRequestInternal(HttpRequest request, ScriptContext context) {
                 buildCookie(cookie);
             }
         }
-        if ("POST".equals(method) || "PUT".equals(method) || "PATCH".equals(method)) {
+        if ("POST".equals(method) || "PUT".equals(method) || "PATCH".equals(method) || "DELETE".equals(method)) {
```

Comment:
you are right. for the time being I'll approve the pul request and tweak more if needed.